### DISCUSSION TITLE
ifdef Doxygen guard in SPI_Master - why???

### DISCUSSION THREAD

**Discussion by FlashForger** [modm-io/modm#1158] [Q&A]
https://github.com/modm-io/modm/blob/f01fd0ab273a2e8180642a0b014c1df13a168968/src/modm/architecture/interface/spi_master.hpp#L33

Why? What for? I am building a project, not a documentation. What for I need a doxygen here?

**Comment by salkinium** [comment]
Some of our interfaces are not "real", they are only there to document the signature of the implementation. However, some signatures are slightly different on different platforms, for example, on AVR we prefer to use smaller integers to pass arguments, since that reduces code size often quite significantly.

In modern times, you would perhaps use C++20 concepts to define them with a bunch of platform specific type aliases. But when we started these interfaces, we didn't have concepts yet.

**Comment by rleh** [comment]
In addition to the answer [above](https://github.com/modm-io/modm/discussions/1158#discussioncomment-9283656): The actual (platform-dependent) implementations of the SPI master is located in `modm/src/modm/platform/spi/*/`, e.g.: https://github.com/modm-io/modm/blob/f01fd0ab273a2e8180642a0b014c1df13a168968/src/modm/platform/spi/stm32/spi_master.hpp.in#L41
